### ISSUE TITLE
Mouse lock problem with dual monitors on Mac

### ISSUE THREAD

**Issue by Uduchi2nd** [ValveSoftware/Dota-2#1211]
I have a dual monitor setup and I play Dota 2 on my right monitor. Whenever I drag my mouse to the far left of the screen, my left monitor picks up on it (I see the OS cursor on the left screen for <1 seconds). It make my edge pan to the left very laggy but all other directions are fine.

I have tried unsuccessfully exclusive full-screen, desktop-friendly full-screen, and turned off and on "Displays have separate spaces" in Mission Control. 

dota_mouse_window_lock 1 is in the console and launch options.

The things that worked: disable the left monitor, or the weird screen arrangement in the attached picture. However, all these things are very annoying to do.

This problem is exclusive to Mac. I didn't have it on Windows or Linux.

![screen shot 2017-04-11 at 11 43 28 am](https://cloud.githubusercontent.com/assets/4780309/24920309/596bc918-1eac-11e7-87e1-62e840a807b0.png)

**Comment by gdrewb-valve** [comment]
This sounds very similiar to #716.

**Comment by kisak-valve** [comment]
Possible duplicate of #1030.

**Comment by brandonshough** [comment]
Is this being looked at? It is super frustrating.

**Comment by mandal-ts** [comment]
There are 2 solution:

1. close the laptop lid
2. use mirror mode on display setting

**Comment by timfitzzz** [comment]
> This sounds very similiar to #716.

Just wanted to note that I believe that I did in fact have a secondary display active, so these are probably the same issue.

**Comment by kierien** [comment]
Have been experiencing this for the past 4 months as well, was working fine prior.

**Comment by nldoty** [comment]
> There are 2 solution:
> 
>     1. close the laptop lid
> 
>     2. use mirror mode on display setting

These are not solutions. These are temporary workarounds.          
If your door on your car no longer opens, "crawl out the window" is not a solution. The door is still broken.

**Comment by kcc0** [comment]
+1

Can someone please fix this?

**Comment by Delitants** [comment]
> +1
> 
> Can someone please fix this?

No, they can't, it's been 3 years and is getting ridiculous. This is the main reason why I stopped playing. They probably don't know how to fix it or being lazy about it.

**Comment by arveev** [comment]
5 years has passed and the issue still exists to this date... Hope that someone is still looking into this.

**Comment by babono** [comment]
same here, would be nice if this issue can be fixed...

**Comment by junlouiegonzales** [comment]
Up

**Comment by rickwillcox** [comment]
Amazing this still exists.

**Comment by Mousaka** [comment]
+1

**Comment by roboticsexpert** [comment]
i had this problem,

go to displays setting, select which monitor you want to play with, and make it as main display and everything will works well

![Image](https://github.com/user-attachments/assets/9655cea3-6ba5-423e-9ec6-5f767f5d9617)
